docs/user/technical-reference/03-10-buildless-serverless.md

```diff
@@ -0,0 +1,48 @@
+# Build-less mode of Serverless
+
+## Overview: Moving Towards Buildless Mode
+
+From the beginning, Kyma Serverless has aimed to accelerate the development of fast prototypes by allowing users to focus on business logic rather than containerization and Kubernetes deployment. Our goal is to remove operational barriers so developers can iterate quickly and efficiently.
+
+With the introduction of Buildless mode, we are taking this vision further. By eliminating the image build step in the Kyma runtime, we significantly shorten the feedback loop during prototype development. In buildless mode, instead of building and pushing custom function images into in-cluster registry, your code and dependencies are simply mounted into Kyma-provided runtime images. This approach positions Kyma Serverless as more efficient development tool, enabling even faster iteration. Additionally, it eliminates the architectural complexities and limitations of deploying serverless Functions on Kubernetes.
+
+## Benefits
+
+- **Faster deployment**: Functions deploy faster as the build job is no longer required.
+- **Resource efficiency**: Eliminates the need to for Serverless to acquire computational resources from worker nodes to build the image.
+- **Enhanced security**: By eliminating build jobs functions can run in namespaces with more restrictive pod security levels enabled.
+- **No additional storage required**: No additional storage resources are used to store the Function image.
+- **Simplified Architecture**: The Serverless module no longer requires a Docker Registry, making it more lightweight and easier to manage.
+
+## What would change if I switch buildless on
+
+- The internal resources used for storing custom function images (Docker Registry) will be uninstalled from the Serverless module
+- Your  Functions will start quicker as build Jobs for Functions are no longer created (and existing Jobs resources will be removed).
+- Libraries and dependencies are downloaded dynamically at the start of each Function's execution.
+- Function code is directly injected into the runtime Pod, removing the need for pre-built images.
+- Your existing Functions will redeploy (without downtime) and will be started as Pods based on Kyma-provided images with your code and dependencies mounted.
+
+## Use fixed dependency versions
+
+- **Avoid using `latest` versions of Function dependencies**: Since dependencies are resolved at Function's Pod start time runtime in build-less mode, using `latest` versions can lead to inconsistencies between replicas of the same Function. This may be the case when dependency provider releases a new version after one replica is already running and before another replica was created due to auto-scaling.  Always specify exact versions of dependencies to ensure stability and predictability.
```

Comment:
```suggestion
- **Avoid using `latest` versions of Function dependencies**: Since dependencies are resolved at Function's Pod start time in buildless mode, using `latest` versions can lead to inconsistencies between replicas of the same Function. This may be the case when the dependency provider releases a new version after one replica is already running and before another replica is created due to auto-scaling.  Always specify exact versions of dependencies to ensure stability and predictability.
```